src/components/ble/SimpleWeatherService.cpp

```diff
@@ -0,0 +1,151 @@
+/*  Copyright (C) 2023 Jean-François Milants
+
+    This file is part of InfiniTime.
+
+    InfiniTime is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    InfiniTime is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "components/ble/SimpleWeatherService.h"
+
+#include <algorithm>
+#include <cstring>
+#include <nrf_log.h>
+
+using namespace Pinetime::Controllers;
+
+namespace {
+  enum class MessageType : uint8_t { CurrentWeather, Forecast, Unknown };
+
+  uint64_t ToUInt64(const uint8_t* data) {
+    return *(reinterpret_cast<const uint64_t*>(data));
+  }
+
+  SimpleWeatherService::CurrentWeather CreateCurrentWeather(const uint8_t* dataBuffer) {
+    char cityName[33];
+    std::memcpy(&cityName[0], &dataBuffer[13], 32);
+    cityName[32] = '\0';
+    return SimpleWeatherService::CurrentWeather {ToUInt64(&dataBuffer[2]),
+                                                 dataBuffer[10],
+                                                 dataBuffer[11],
+                                                 dataBuffer[12],
+                                                 dataBuffer[13 + 32],
+                                                 cityName};
```

Comment:
It would be best to use parentheses to make it clear that we aren't using struct initialisation, but the constructor.

```suggestion
    return SimpleWeatherService::CurrentWeather(ToUInt64(&dataBuffer[2]),
                                                dataBuffer[10],
                                                dataBuffer[11],
                                                dataBuffer[12],
                                                dataBuffer[13 + 32],
                                                cityName);
```

Reply:
Done